simpa/core/simulation_modules/reconstruction_module/reconstruction_utils.py

```diff
@@ -412,6 +403,31 @@ def preparing_reconstruction_and_obtaining_reconstruction_settings(
             time_spacing_in_ms, torch_device)
 
 
+def get_reconstruction_processing_unit(global_settings):
+    """
+    Get device (CPU/GPU) for reconstructing the image.
+    :param global_settings: global SIMPA settings
+    :return: device for reconstruction
+    """
+    logger = Logger()
+    # if no tag is set, try to use GPU if available
+    if Tags.GPU not in global_settings:
+        if torch.cuda.is_available():
+            dev = "cuda"
+        else:
+            dev = "cpu"
+    else:  # else set tag as user wants
+        dev = "cuda" if global_settings[Tags.GPU] else "cpu"
+
+    if dev == 'cuda' and not torch.cuda.is_available():
+        # torch will likely raise an error if no GPU is available but set as device -> log it to SIMPA log
+        logger.error('Cuda is not available! Check your torch/cuda version.')
```

Comment:
Would it not make sense to log a warning and default to dev="cpu" afterwards?
Otherwise, the simulations might crash. Not sure what is preferable..

Reply:
I discussed with @TomTomRixRix and probably it is better if the programm crashs (if user set GPU tag) and the user than can choose to set cpu as processing unit OR check why cuda is not available.

Reply:
I think it depends on the use case. If you want to reconstruct images and it takes 3 hours on CPU instead of 10 seconds on GPU to reconstruct a volume than letting the program crash and restart it with proper settings saves you time. For other use cases where the difference between GPU and CPU is not that large, resorting to CPU might not be that much of a problem

Reply:
You are right. But the crash case, is the case where to user decided to set the GPU Tag. So the user expects that there is a GPU and that it is working. So I guess we can point (not just poke like logging ; ) ) the user to a problem that the user might not have been aware of.